graph/list.go

```diff
@@ -108,17 +108,15 @@ func (s *TagStore) Images(filterArgs, filter string, all bool) ([]*types.Image,
 			} else {
 				// get the boolean list for if only the untagged images are requested
 				delete(allImages, id)
-				if !imageFilters.MatchKVList("label", image.ContainerConfig.Labels) {
+				if len(imageFilters["label"]) > 0 && image.Config == nil {
```

Comment:
This check is for "If we are filtering on labels and the image doesn't contains `Config` struct, let's skip it — as it will not have Labels". I don't like this because if we introduce new filters, the check is going to be ugly (or refactor will be needed).